src/parsers/enum_destructuring.cr

```diff
@@ -11,7 +12,16 @@ module Mint
 
         option = type_id! EnumDestructuringExpectedOption
 
-        parameters = many { type_variable }.compact
+        parameters = (many do
```

Comment:
I think we should have only one syntax, in this case with parentheses. This will be a breaking change, but would fix the else branch since there it's not possible to inner destructure.

Reply:
Maybe we want to have a comma between parameters?